compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaParser.java

```diff
@@ -5071,6 +5076,13 @@ private STNode parseTerminalExpression(STNode annots, List<STNode> qualifiers, b
         parseExprQualifiers(qualifiers);
         STToken nextToken = peek();
 
+        STNodeList annotNodeList = (STNodeList) annots;
+        if (!annotNodeList.isEmpty() && !isAnnotAllowedExprStart(nextToken)) {
+            annots = addAnnotNotAttachedDiagnostic(annotNodeList);
+            STNode qualifierNodeList = createObjectTypeQualNodeList(qualifiers);
+            return createMissingObjectConstructor(annots, qualifierNodeList);
+        }
```

Comment:
This will give misleading errors for samples like below right?
```bal
public function main() {
    int a = 43;
    int b = @deprecated a;
}
```
Currently, this gives `annotations attached to expression` which is correct but with this change, it will give many errors which mislead the user. I think recovering as object cons when annot is nonempty rather than using `isAnnotAllowedExprStart(nextToken)` will fix this issue up to some extend WDYT?

Reply:
Changes made as per our offline discussion